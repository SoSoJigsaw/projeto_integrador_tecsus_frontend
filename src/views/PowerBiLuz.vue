<template>
  <transition name="fade" mode="out-in">
    <div v-if="isVisible" class="container-powerbi">
      <h1>Dashboard de Luz</h1>
      <!-- <iframe></iframe>  -->
    </div>
  </transition>
</template>

<script lang="ts">
export default {
  data() {
    return {
      isVisible: false
    }
  },

  mounted() {
    setTimeout(() => {
      this.isVisible = true
    }, 1000)
  }
}
</script>

<style>
@import '../assets/base.css';

.container-powerbi {
  margin-left: 4%;
  margin-top: 3%;
  margin-right: 4%;
  margin-bottom: 3%;
  display: flex;
  flex-direction: column;
  gap: 4.5rem;
}

.container-powerbi h1 {
  color: #776ae3;
  font-size: 6vmin;
  font-weight: 400;
}
</style>
