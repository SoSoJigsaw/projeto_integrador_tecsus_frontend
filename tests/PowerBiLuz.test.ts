import { describe, it, expect, vi } from 'vitest'
import { mount, flushPromises } from '@vue/test-utils';
import PowerBiLuz from '../src/views/PowerBiLuz.vue'

const createWrapper = (isVisible = true, powerBiSrc = '') => {
  return mount(PowerBiLuz, {
    data() {
      return { isVisible, powerBiSrc };
    } 
  });
};

describe('PowerBiLuz.vue', () => {

  it('should mount the component correctly', async () => {
    const wrapper = createWrapper();
    await flushPromises();
    expect(wrapper.exists()).toBe(true)
  })

  it('should start with isVisible set to false', async () => {
    const wrapper = createWrapper(false);
    await flushPromises();
    expect(wrapper.vm.isVisible).toBe(false)
  })

  it('should display the main container after 1 second', async () => {
    vi.useFakeTimers()
    const wrapper = createWrapper();
    vi.advanceTimersByTime(1000)
    await flushPromises();
    expect(wrapper.find('.container-powerbi').isVisible()).toBe(true)
    vi.useRealTimers()
  })

  it('should render iframe with the correct Power BI source', async () => {
    const wrapper = createWrapper();
    await flushPromises();
    const iframe = wrapper.find('iframe')
    expect(iframe.exists()).toBe(true)
    expect(iframe.attributes('src')).toBe(wrapper.vm.powerBiSrc)
  })

  it('should render without errors with default props', async () => {
    const wrapper = createWrapper(false);
    await flushPromises();
    expect(wrapper.html()).toMatchSnapshot()
  })

});