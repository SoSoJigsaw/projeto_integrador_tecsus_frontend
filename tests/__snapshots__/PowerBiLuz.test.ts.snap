// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PowerBiLuz.vue > should render without errors with default props 1`] = `"<div class="entre-paginas"><img src="/projeto_integrador_tecsus_frontend/src/assets/icons/luz.png"></div>"`;
